File: JAVA-Basic/JavaBasic/src/ConditionalStatementsAdvanced/Exercise/FishingBoat.java
package ConditionalStatementsAdvanced.Exercise;

import java.util.Scanner;

public class FishingBoat {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int budget = Integer.parseInt(scan.nextLine());
        String season = scan.nextLine();
        int fisherMens = Integer.parseInt(scan.nextLine());
        double shipPrice = 0.0;
        boolean discount = true;

        switch (season) {
            case "Spring":
                shipPrice = 3000;
                break;
            case "Summer":
                shipPrice = 4200;
                break;
            case "Autumn":
                shipPrice = 4200;
                discount = false;
                break;
            case "Winter":
                shipPrice = 2600;
                break;
        }
        if (fisherMens <= 6) {
            shipPrice *= 0.9;
        } else if (fisherMens <= 11) {
            shipPrice *= 0.85;
        } else if (fisherMens > 11) {
            shipPrice *= 0.75;
        }
        if (fisherMens % 2 == 0 && discount) {
            shipPrice *= 0.95;
        }
        if (budget >= shipPrice) {
            System.out.printf("Yes! You have %.2f leva left.", budget - shipPrice);
        } else {
            System.out.printf("Not enough money! You need %.2f leva.", shipPrice - budget);
        }
    }
}

File: 01.JS-Basic/По сложни проверки Упражнение/FishingBoat.js
function fishingBoat(input) {

    let budget = Number(input[0])
    let season = input[1]
    let fishermans = Number(input[2])    //"Spring", "Summer", "Autumn", "Winter"

    let price = 0.0

    switch (season) {
        case "Spring":
            if (fishermans <= 6) {
                price = 3000 * 0.9
            }
            else if (fishermans > 7 && fishermans <= 11) {
                price = 3000 * 0.85
            }
            else if (fishermans > 12) {
                price = 3000 * 0.75
            }; break;
        case "Summer":
            if (fishermans <= 6) {
                price = 4200 * 0.9
            }
            else if (fishermans > 7 && fishermans <= 11) {
                price = 4200 * 0.85
            }
            else if (fishermans > 12) {
                price = 4200 * 0.75
            }; break;
        case "Autumn":
            if (fishermans <= 6) {
                price = 4200 * 0.9
            }
            else if (fishermans > 7 && fishermans <= 11) {
                price = 4200 * 0.85
            }
            else if (fishermans > 12) {
                price = 4200 * 0.75
            }; break;
        case "Winter":
            if (fishermans <= 6) {
                price = 2600 * 0.9
            }
            else if (fishermans > 7 && fishermans <= 11) {
                price = 2600 * 0.85
            }
            else if (fishermans > 12) {
                price = 2600 * 0.75
            }; break;
    }
    if (fishermans % 2 == 0 && season != "Autumn") {
        price = price * 0.95
    }
    if (budget >= price) {
        console.log(`Yes! You have ${(budget - price).toFixed(2)} leva left.`)
    }
    else { console.log(`Not enough money! You need ${(price - budget).toFixed(2)} leva.`) }
}
fishingBoat(["3600",
    "Autumn",
    "6"])


